test/integration/lib/lockfile.spec.ts

```diff
@@ -1,196 +1,200 @@
 import { expect } from 'chai';
-import { dirname, basename } from 'path';
-import { stub, SinonStub } from 'sinon';
-import { promises as fs } from 'fs';
-import mock = require('mock-fs');
-
+import { promises as fs, mkdirSync } from 'fs';
+import { testfs, TestFs } from 'mocha-pod';
+import * as os from 'os';
+import * as path from 'path';
+import { stub } from 'sinon';
 import * as lockfile from '~/lib/lockfile';
 import * as fsUtils from '~/lib/fs-utils';
-const BASE_LOCK_DIR = '/tmp/balena-supervisor/services';
-const LOCKFILE_UID = 65534;
 
-describe('lib/lockfile', () => {
-	const lockPath = `${BASE_LOCK_DIR}/1234567/one/updates.lock`;
-	const lockPath2 = `${BASE_LOCK_DIR}/7654321/two/updates.lock`;
-
-	const mockDir = (opts: { createLock: boolean } = { createLock: false }) => {
-		mock({
-			[BASE_LOCK_DIR]: {
-				'1234567': {
-					one: opts.createLock
-						? { 'updates.lock': mock.file({ uid: LOCKFILE_UID }) }
-						: {},
-				},
-				'7654321': {
-					two: opts.createLock
-						? { 'updates.lock': mock.directory({ uid: LOCKFILE_UID }) }
-						: {},
-				},
-			},
-		});
-	};
-
-	const checkLockDirFiles = async (
-		path: string,
-		opts: { shouldExist: boolean } = { shouldExist: true },
-	) => {
-		const files = await fs.readdir(dirname(path));
-		if (opts.shouldExist) {
-			expect(files).to.include(basename(path));
-		} else {
-			expect(files).to.have.length(0);
-		}
-	};
-
-	let execStub: SinonStub;
-
-	beforeEach(() => {
-		// @ts-ignore
-		execStub = stub(fsUtils, 'exec').callsFake(async (command, opts) => {
```

Comment:
Check this out @cywang117, we no longer need to mock `exec` in the lockfile tests when using `mocha-pod`

Reply:
Nice~ excited to check this out. Looking forward to rewriting the API tests with mocha-pod!  I will close my mock-fs utility branch from a while ago 😅 (it's local but it's a utility to mock the entire filesystem, mocha-pod just seems better)